File: Calculadora_de_ceros_de_una_ecuación_cuadrática.py
from math import sqrt

def igualAUno(a, b, c):
    discriminante = pow((b/2), 2) - c

    if discriminante < 0:
        print("La ecuación no tiene soluciones reales")
        x1 = complex(-b/2, sqrt(-discriminante))
        x2 = complex(-b/2, sqrt(-discriminante))
    else:
        x1 = (-b/2) + sqrt(discriminante)
        x2 = (-b/2) - sqrt(discriminante)

    print("La primera raiz es " + str(x1))
    print("La segunda raiz es " + str(x2))

def difenteDeUno(a, b, c):
    discriminante = pow(b, 2) - 4 * a * c

    if discriminante < 0:
        print("La ecuación no tiene soluciones reales")
        x1 = complex((-b + sqrt(-discriminante)) / (2 * a), 0)
        x2 = complex((-b - sqrt(-discriminante)) / (2 * a), 0)
    else:
        x1 = (-b + sqrt(discriminante)) / (2 * a)
        x2 = (-b - sqrt(discriminante)) / (2 * a)

    print("La primera raiz es " + str(x1))
    print("La segunda raiz es " + str(x2))
    
def main():
    print("Hola esta es una calculadora para calcular las raices de una ecuación de segundo grado")
    a = float(input("Ingresa a: "))
    b = float(input("Ingresa b: ")) 
    c = float(input("Ingresa c: "))

    if a == 1:
        igualAUno(a, b, c)
    else:
        difenteDeUno(a, b, c)

if __name__ == "__main__":
    main()

File: Calculadora_de_ceros_de_una_ecuación_cuadrática.cs
using System;

class Program
{
    static void igualAUno(double a, double b, double c)
    {
        double discriminante = Math.Pow((b / 2), 2) - c;

        if (discriminante < 0)
        {
            Console.WriteLine("La ecuación no tiene soluciones reales");
            double x1 = (-b / 2) + Math.Sqrt(-discriminante);
            double x2 = (-b / 2) - Math.Sqrt(-discriminante);
            Console.WriteLine("La primera raiz es " + x1 + "i");
            Console.WriteLine("La segunda raiz es " + x2 + "i");
        }
        else
        {
            double x1 = (-b / 2) + Math.Sqrt(discriminante);
            double x2 = (-b / 2) - Math.Sqrt(discriminante);
            Console.WriteLine("La primera raiz es " + x1);
            Console.WriteLine("La segunda raiz es " + x2);
        }
    }

    static void diferenteDeUno(double a, double b, double c)
    {
        double discriminante = Math.Pow(b, 2) - 4 * a * c;

        if (discriminante < 0)
        {
            Console.WriteLine("La ecuación no tiene soluciones reales");
            double x1 = ((-b) + Math.Sqrt(-discriminante)) / (2 * a);
            double x2 = ((-b) - Math.Sqrt(-discriminante)) / (2 * a);
            Console.WriteLine("La primera raiz es " + x1 + "i");
            Console.WriteLine("La segunda raiz es " + x2 + "i");
        }
        else
        {
            double x1 = ((-b) + Math.Sqrt(discriminante)) / (2 * a);
            double x2 = ((-b) - Math.Sqrt(discriminante)) / (2 * a);
            Console.WriteLine("La primera raiz es " + x1);
            Console.WriteLine("La segunda raiz es " + x2);
        }
    }

    static void Main()
    {
        char opc;
        Console.WriteLine("----------------- Hola esta es una calculadora para calcular las raices de una ecuación de segundo grado -----------------");

        do
        {
            Console.Write("Ingresa el valor de a: ");
            double a = double.Parse(Console.ReadLine());
            Console.Write("Ingresa el valor de b: ");
            double b = double.Parse(Console.ReadLine());
            Console.Write("Ingresa el valor de c: ");
            double c = double.Parse(Console.ReadLine());

            if (a == 1)
            {
                igualAUno(a, b, c);
            }
            else
            {
                diferenteDeUno(a, b, c);
            }
            Console.Write("¿Quieres continuar? S/N: ");
            opc = Console.ReadLine()[0];
        } while (opc != 'N' && opc != 'n');

        Console.WriteLine("Presiona cualquier tecla para salir");
        Console.ReadKey();
    }
}